cdn/signed-urls/src/main/java/com/google/cdn/SignedUrls.java

```diff
@@ -75,6 +76,61 @@ public static String getSignature(byte[] privateKey, String input)
   }
   // [END cloudcdn_sign_url]
 
+  // [START cloudcdn_sign_url_prefix]
+  /**
+   * Creates a signed URL with a URL prefix for a Cloud CDN endpoint with the given key. Prefixes
+   * allow access to any URL with the same prefix, and can be useful for granting access broader
+   * content without signing multiple URLs.
+   *
+   * @param requestUrl URL of request
+   * @param urlPrefix URL prefix to sign as a string. urlPrefix must start with either http:// or
+   * https:// and should not include query parameters
+   * @param key url signing key uploaded to the backend service/bucket, as a 16-byte array
+   * @param keyName the name of the signing key added to the back end bucket or service
+   * @param expirationTime the date that the signed URL expires
+   * @return a properly formatted signed URL with URL prefix
+   * @throws InvalidKeyException when there is an error generating the signature for the input key
+   * @throws NoSuchAlgorithmException when HmacSHA1 algorithm is not available in the environment
+   * @throws IllegalArgumentException when urlPrefix string is malformed.
+   */
+  static String signUrlWithPrefix(String requestUrl,
+      String urlPrefix,
+      byte[] key,
+      String keyName,
+      Date expirationTime)
+      throws InvalidKeyException, NoSuchAlgorithmException, IllegalArgumentException {
+
+    if (urlPrefix.contains("?") || urlPrefix.contains("#")) {
+      throw new IllegalArgumentException("urlPrefix must not include query params: " + urlPrefix);
+    }
+    if (!urlPrefix.startsWith("http://") && !urlPrefix.startsWith("https://")) {
```

Comment:
```suggestion
    if (!urlPrefix.startsWith("http://")) {
```

Reply:
Since we only need to check if the urlPrefix does not start with "http://", there is no need to explicitly check for "https://" as well.
The second condition will be captured as part of the first condition.

Reply:
> The second condition will be captured as part of the first condition.

I disagree, both condition are needed. With only the first condition,  `https://` will be falsely identified as invalid. Simply try`assertFalse("https://".startsWith("http://"));`
Unless you suggest to capture invalid cases with _"not starting with `http`"_, which still is not the same as saying "only starting with `http://` and `https://` are valid"

Reply:
Apologies for the confusion. The suggestion was to make it`(!urlPrefix.startsWith("http")` which would capture both the conditions.
Unless you have to explicitly put both conditions in there, might be okay to change it.

Reply:
Actually on a second thought, I think it's better to demonstrate 2 conditions instead of one. That way the customer knows the URL can start with either "http" or "https".
By simplifying the condition, we'll be only checking if it starts with "http".

Hence, resolving this comment. Thank you.